episodes/anonymization.md

```diff
@@ -32,19 +33,352 @@ Metadata elements in imaging, such as patient names and addresses, are often cle
+    else:
+        # just move over a bit and hope for better
+        new_possible_point = round(img.GetSize()[0]/2) + round(img.GetSize()[0]/10) , round(img.GetSize()[1]/2), round(img.GetSize()[2]/2)
+        picked_point = new_possible_point
+    return picked_point
+# do some reality check of a look at the value in your seed point  
+seed_point = guess_seed_point(sag_image) 
+pixel_value = sag_image.GetPixel(seed_point)
+print(pixel_value)
+ ```
+
+ ```output
+ 394.37042236328125
+ ```
+
+Grow from seed, then display:
+
+
+ ```python
+seed_point = guess_seed_point(sag_image)  
+lower_threshold = 370   # lower threshold
+upper_threshold = 480   # upper threshold
+
+
+seed_mask = sitk.ConnectedThreshold(
+    sag_image, seedList=[seed_point],
+    lower=lower_threshold,
+    upper=upper_threshold)
+# let's dilate up a bit
+seed_mask= sitk.BinaryDilate(seed_mask, [5, 5, 5])
+
+# apply the mask to the image
+brain_only = sitk.Mask(sag_image, seed_mask)
+# display to see what happened
+interact(
+    display_images,
+    image1_z=(0, sag_image.GetSize()[2] - 1),
+    image2_z=(0, brain_only.GetSize()[2] - 1),
+    image1_npa=fixed(sitk.GetArrayViewFromImage(sag_image)),
+    image2_npa=fixed(sitk.GetArrayViewFromImage(brain_only)),
+    title1=fixed("Original"),
+    title2=fixed("Seeded and Masked")
+)
+ ```
+
+
+ ```output
+```
+
+![Our grown from seed soft tissue stripping.](fig/grown_from_seed.png){alt='Home-made deface by grow from seed'}
+
+ In the two solutions above we made masks. They each have different problems we can see. For example in the grown from seed images we stripped out part of the genu and splenium of the corpus callosum not to mention the entire pons. An alternative is that could also have used a registered atlas as a mask. Regardless of the base algorithms we begin with, if we want a better image, we can keep tinkering with our algorithms.
+
+:::::::::::::::::::::::::
+
+
+::::::::::::::::::::::::::::::::::::::::::::::::::
+
+In the provided solutions to the above optional challenge we didn't get rid of a lot of the neck, and had other issues. A different approach we could have taken would have been to register the brain to a brain-like shape and use this as a mask. Nonetheless the given solutions shows two potential approaches to removing tissue (which could in some cases lead to identification) you don't want in an image. Many researchers do not want to waste time optimizing a skull stripping technique if this was not the research question, so they use a pre-made technique.
+
+
+There are various tools available for defacing head imaging, ranging from fully developed software products like [FreeSurfer](https://surfer.nmr.mgh.harvard.edu/), which includes built-in defacing capabilities, to specialized functions within coding libraries. Some of these tools strip off all of the skull and soft tissue which may be useful for analysis even if we don't care about deidentification e.g. if we only want to look at brain tissue.  
```

Comment:
These tools give a whole new meaning to the expression "putting your nose to the grindstone". Good thing it'st just your digital twin undergoing the procedure.